src/internal/hooks/use-dynamic-overlap/__tests__/use-dynamic-overlap.test.tsx

```diff
@@ -1,10 +1,10 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
-import React from 'react';
+import React, { useState } from 'react';
 import { render, screen } from '@testing-library/react';
 
-import { useDynamicOverlap } from '../../../../../lib/components/app-layout/visual-refresh/hooks/use-dynamic-overlap';
-import { AppLayoutContext, AppLayoutProvider } from '../../../../../lib/components/app-layout/visual-refresh/context';
```

Comment:
Moved this file from `src/app-layout/hooks` into `src/internal/hooks` because it is in fact a cross-component API

Also look how it helped to remove dependency on app layout internals. It now uses only the package-public context